src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +13,260 @@ public static class NUnitTestExtractorApp
+        private static HashSet<string> getValidTestsFromTestSuite(Type testSuite)
         {
-            if (!string.IsNullOrEmpty(excludeInfo))
+            HashSet<string> tests = new HashSet<string>();
+
+            foreach (MethodInfo test in testSuite.GetMethods())
             {
-                excludeInfo = excludeInfo.Replace("\"", "");
+                var attributes = test.GetCustomAttributesData();
 
-                List<string> excludeInfoValues = new List<string>(excludeInfo.Split(','));
+                // search function attributes to find out whether the function has "Test" or "TestCase" attribute defined
+                if (!attributes.Any(x => x.AttributeType.Equals(typeof(TestAttribute)))
+                    && !attributes.Any(x => x.AttributeType.Equals(typeof(TestCaseAttribute))))
+                {
+                    continue;
+                }
+
+                // search the category attributes for the function
+                var categories = attributes.Where(x => x.AttributeType.Equals(typeof(CategoryAttribute)));
 
-                foreach (string value in excludeInfoValues)
+                // match the "Exclude" category from the function category list
+                // if the function has matched exclude category, we will skip this function
+                if (categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.ExcludeInfo))))
                 {
-                    if (categoryNames != null && categoryNames.Contains(value))
-                    {
-                        return true;
-                    }
+                    continue;
                 }
-                return false;
-            }
-            throw new ArgumentNullException("excludeInfo", "no excludeInfo specified");
-        }
 
-        /// <summary>
-        /// Depending on number of args within the TestCase, function will return properly formatted testcase
-        /// </summary>
-        /// <param name="declaringType">declaring type of testcase method(namespace.class)</param>
-        /// <param name="name">name of testcase</param>
-        /// <param name="testCaseArgs">the object array containing the testcase's arguments</param>
-        /// <param name="data">the string which is appended to</param>
-        /// <returns>the properly formatted TestCase string</returns>
-        public static string FormattedTestCase(string declaringType, string name, object[] testCaseArgs, string data)
-        {
+                // match the "Include" category from the function category list
+                // if the function has matched include category, we will count this function as a valid test
+                if (string.IsNullOrWhiteSpace(_options.IncludeInfo)
+                    || categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.IncludeInfo))))
+                {
+                    Console.WriteLine("Found the test and add it.");
```

Comment:
[Must] Nothing should be written to stdout except "DLL path | test name".

Reply:
This is just for ME - debug purpose :)